### DISCUSSION TITLE
Configurando azure no keycloack

### DISCUSSION THREAD

**Discussion by wbezerra81** [keycloak/keycloak#29169] [Q&A]
### Description

Good Morning!!
Help me please 

I need to configure Azure AD in Keycloack and I'm a little lost because everything is very new to me in this tool.

**Comment by codespearhead** [comment]
What do you mean by "I need to configure Azure AD in Keycloack"?

**Reply by codespearhead** [reply]
@wbezerra81 Keycloak and Azure AD are mutually exclusive, but it's possible to integrate them to an extend, although it brings complexity to the table. That's why it's necessary to understand what you're trying to accomplish, because depending on your goals, you won't need to go through the hassle of having to deal and maintain both of them.

Either way, here's how to [configure Microsoft Azure AD as a Keycloak Identity Provider](https://www.hcl-software.com/blog/versionvault/how-to-configure-microsoft-azure-active-directory-as-keycloak-identity-provider-to-enable-single-sign-on-for-hcl-compass).

**Comment by gilvansfilho** [comment]
Are you trying to use Azure AD as Identity Provider? If yes first you need to check with Azure AD admin which protocol are being used to provide user informations. SAML or OIDC. With that response you will need to setup a according identity provider.

**Comment by wbezerra81** [comment]
Good morning!
I really don't know how to tell you, but I'm going to look for information, but if I had both procedures it would help, because I leave it on file

**Comment by wbezerra81** [comment]
This is the first time I'm using and learning the tool.

**Comment by wbezerra81** [comment]
Could you clear up another doubt for me?

I have a lot of doubts about which Root URL, I get lost a lot on this issue.
I've watched videos on how to configure the URLs but I still don't understand.

**Reply by codespearhead** [reply]
We still don't know what you mean by "configure Azure AD in Keycloack", nor do we know what protocol (REST or SAML) you're using.

Your questions seem to revolve around the OAuth2 standard rather than Keycloak or Azure AD.
